### PR TITLE
chore: Add quickstart guide

### PR DESCRIPTION
**Pull request by makkes**
**What this PR does / why we need it**:
As a newbie to kubefed, it's not easy to start playing around with
kubefed due to the plethora of options on how to deploy it in the user
and development guides. Thus, an opinionated quickstart guide will
help newcomers get up and running in a jiffy.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
no issue

**Special notes for your reviewer**:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,6 +63,52 @@ higher-level APIs:
 
 ## Guides
 
+### Quickstart
+
+1. Clone this repo:
+   ```
+   git clone https://github.com/kubernetes-sigs/kubefed.git
+   ```
+1. Start a kind cluster:
```
Comment (xunpan) [inline README.md:72]: suggest to add `kind` link for users to install `kind` before use it.

**Review by xunpan — COMMENTED** [review]
It is glad to see this quickstart to me. It is now clear to have a quick try to kubefed.

File: README.md (hunk 1 above)
Comment (makkes) [reply README.md]: Great idea! Added that.

File: README.md [hunk 2]
```diff
@@ -63,6 +63,52 @@ higher-level APIs:
...
+   ```
+   kubectl create ns federate-me
+   ```
+1. Tell kubefed to federate that namespace (and the resources in it):
+   ```
+   ./bin/kubefedctl federate ns federate-me
+   ```
+1. Create a `ConfigMap` to be federated:
+   ```
+   kubectl -n federate-me create cm federated-cm
```
Comment (hectorj2f) [inline README.md:101]: Nit: This naming can be confusing `federated-cm`. I would normally create a cm and then federate it, or simply create a federatedConfigMap object.

File: README.md [hunk 3]
```diff
@@ -63,6 +63,52 @@ higher-level APIs:
...
+   ./scripts/fix-joined-kind-clusters.sh
+   ```
+1. Create a namespace to be federated:
+   ```
+   kubectl create ns federate-me
+   ```
+1. Tell kubefed to federate that namespace (and the resources in it):
+   ```
+   ./bin/kubefedctl federate ns federate-me
+   ```
```
Comment (hectorj2f) [inline README.md:98]: You could check if the namespace was propagated.

Comment (hectorj2f) [reply README.md:98]: Also, we are not mentioning anything about placement definitions or explaining them.

Comment (makkes) [reply README.md:98]: > Also, we are not mentioning anything about placement definitions or explaining them.

Given the goal of this guide being providing as little steps as possible to try out kubefed I don't see a need to mention this here. The more advanced docs are explaining that pretty well already.

File: README.md (hunk 2 above)
Comment (makkes) [reply README.md]: sounds good. I changed the CM name.

**Comment by makkes** [comment]
@hectorj2f I addressed your comments.

@xunpan would you mind lgtm'ing/approving again, please?

**Comment by xunpan** [comment]
/lgtm

Thanks

**Pull request merged**